spec/services/forest_liana/filter_parser_spec.rb

```diff
@@ -0,0 +1,272 @@
+module ForestLiana
+  describe FilterParser do
+    let(:timezone) { 'Europe/Paris' }
+    let(:resource) { Tree }
+    let(:filters) { {} }
+    let(:filter_parser) { described_class.new(filters.to_json, resource, timezone) }
+    let(:simple_condition_1) { { 'field' => 'name', 'operator' => 'contains', 'value' => 'Tree' } }
+    let(:simple_condition_2) { { 'field' => 'name', 'operator' => 'ends_with', 'value' => '3' } }
+    let(:simple_condition_3) { { 'field' => 'age', 'operator' => 'greater_than', 'value' => 2 } }
+    let(:belongs_to_condition) { { 'field' => 'trees:age', 'operator' => 'less_than', 'value' => 3 } }
+    let(:date_condition_1) { { 'field' => 'created_at', 'operator' => 'before', 'value' => Time.now - 2.hours } }
+    let(:date_condition_2) { { 'field' => 'created_at', 'operator' => 'today' } }
+    let(:date_condition_3) { { 'field' => 'created_at', 'operator' => 'previous_x_days', 'value' => 3 } }
+
+    before {
+      island = Island.create!(name: "L'île de la muerta")
+      king = User.create!(title: :king)
+      villager = User.create!(title: :villager)
+      Tree.create!(name: 'Tree n1', age: 1, island: island, owner: king)
+      Tree.create!(name: 'Tree n2', age: 3, island: island, created_at: Time.now - 1.day, owner: king)
+      Tree.create!(name: 'Tree n3', age: 4, island: island, owner: king, cutter: villager)
+    }
+
+    after {
+      Tree.destroy_all
+      User.destroy_all
+      Island.destroy_all
+    }
+
+    describe 'apply_filters' do
+      context 'on valid filters' do
+        let(:filters) { { aggregator: 'and', conditions: [simple_condition_1, simple_condition_2] } }
+        it { expect(filter_parser.apply_filters.count).to eq 1 }
+      end
+    end
+
+    describe 'parse_aggregator' do
+      let(:query) { filter_parser.parse_aggregator(filters) }
+
+      context 'when no aggregator' do
+        let(:filters) { simple_condition_1 }
+        it { expect(resource.where(query).count).to eq 3 }
+      end
+
+      context "simple_condition_1 'and' simple_condition_2" do
```

Comment:
Does not look to be very human readable/descriptive tests. I think you should mention what kind of filter condition you test to make it easy to understand what breaks if a specific test fails.

`error on "simple_condition_1 'and' simple_condition_2"` is not easy to read for instance